text/0000-add-scratch-stack.md

```diff
@@ -0,0 +1,161 @@
+# Create scratch stack
+
+## Summary
+
+The Paketo stacks maintainers should create and maintain a scratch stack that
+can run statically-linked applications (e.g. golang, Rust). The Builder team does not
+need to create a builder for this image, but may do so if they wish.
+
+Additionally, the golang buildpack should gain support for this new scratch
+stack, both in order to provide a useful experience to golang application
+developers as well as to enable integration tests for this new stack.
+
+## Motivation
+
+Some applications consist of a single
+[statically-linked](https://en.wikipedia.org/wiki/Static_library) executable.
+This is most common in golang, but is also often utilized in rust. It is also
+common in languages/toolchains not supported by Paketo like C/C++. These
+statically-linked binaries can be run on a minimally-modified empty container
+image (AKA "scratch" image) with few-to-no packages/libraries. Scratch images
+are a common choice of container image for statically-linked executables as it
+minimizes the size (in MB) of the image as well as the number of installed
+packages - in turn reducing the surface area for security issues.
+
+Currently, in order to build these statically-linked applications using Paketo
+buildpacks (and aiming for the smallest possible resultant run image) users
+must either opt for the [tiny
+stack](https://github.com/paketo-buildpacks/jammy-tiny-stack) or build their
+own scratch stack. We have evidence of users doing both of these things.
+Neither are satisfactory to users, however, as the tiny stack contains more
+packages than is necessary for those applications, and users would prefer not
+to maintain their own stacks if Paketo is willing and able to do so.
```

Comment:
I would argue though, that maintaining a scratch stack is not that heavy of a burden - considering there are few to no packages in it. By that same reasoning, as a stacks maintainer I'm probably fine with adopting a scratch stack in the project.

Reply:
I think it is somewhat unreasonable to ask Paketo consumers who want to build the smallest-possible images from their statically-linked golang/rust applications to maintain their own stack and builder for this purpose. Application developers shouldn't need to know about stacks and builders to be able to use `pack build` (or equivalent).

The corollary of that is that it is a significantly lower burden for stacks maintainers to create the stack and keep it up to date - as you point out.